lib/Language/Souffle/Interpreted.hs

```diff
@@ -178,47 +191,62 @@ instance MonadSouffle SouffleM where
       souffleTempDir <- liftIO $ do
         tmpDir <- getCanonicalTemporaryDirectory
         createTempDirectory tmpDir "souffle-haskell"
-      let factDir = souffleTempDir </> "fact"
-          outDir  = souffleTempDir </> "out"
+
+      factDir <- fromMaybe (souffleTempDir </> "fact") <$> asks cfgFactDir
+      outDir <- fromMaybe (souffleTempDir </> "out") <$> asks cfgOutputDir
       liftIO $ do
         createDirectoryIfMissing True factDir
         createDirectoryIfMissing True outDir
       mSouffleBin <- asks cfgSouffleBin
       liftIO $ forM mSouffleBin $ \souffleBin ->
-        fmap Handle $ newIORef $ HandleData
-          { soufflePath = souffleBin
-          , basePath    = souffleTempDir
-          , factPath    = factDir
-          , outputPath  = outDir
-          , datalogExec = datalogExecutable
-          , noOfThreads = 1
-          }
+        Handle <$> ((,,)
+          <$> (newIORef $ HandleData
```

Comment:
These 2 lines become much simpler if the Handle newtype is changed to data like mentioned before.